components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/api/model/subscription/Subscription.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.webhook.management.api.model.subscription;
+
+/**
+ * Represents a subscription to a webhook channel.
+ */
+public class Subscription {
+
+    private SubscriptionStatus status;
+    private String channelUri;
+
+    private Subscription(Builder builder) {
+
+        this.status = builder.status;
+        this.channelUri = builder.channelUri;
+    }
+
+    public SubscriptionStatus getStatus() {
+
+        return status;
+    }
+
+    public void setStatus(SubscriptionStatus status) {
+
+        this.status = status;
+    }
```

Comment:
why do we have setters when there's a builder

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/fb575c056228c93d2221b4ea29c500a76d09311c